### DISCUSSION TITLE
[@types/ramda]: evolve only accepts same type as input

### DISCUSSION THREAD

**Discussion by assertnotnull** [DefinitelyTyped/DefinitelyTyped#60727] [Issues with a @types package]
The current typing is not allowing to do transformations as such as:

https://ramdajs.com/repl/?v=0.28.0#?const%20tomato%20%3D%20%7BfirstName%3A%20%27%20%20Tomato%20%27%2C%20data%3A%20%7Belapsed%3A%20100%2C%20remaining%3A%201400%7D%2C%20id%3A123%7D%3B%0Aconst%20transformations%20%3D%20%7B%0A%20%20firstName%3A%20%28val%29%20%3D%3E%20%28%7Bstruct%3A%20R.trim%28val%29%7D%29%2C%0A%20%20lastName%3A%20%28val%29%20%3D%3E%20R.trim%2C%20%2F%2F%20Will%20not%20get%20invoked.%0A%20%20data%3A%20%7Belapsed%3A%20R.add%281%29%2C%20remaining%3A%20R.add%28-1%29%7D%0A%7D%3B%0AR.evolve%28transformations%2C%20tomato%29%3B%20%2F%2F%3D%3E%20%7BfirstName%3A%20%27Tomato%27%2C%20data%3A%20%7Belapsed%3A%20101%2C%20remaining%3A%201399%7D%2C%20id%3A123%7D

Where evolve is used to change structure. 
`Evolver<MyObj>` doesn't accept functions that return a diffrent type than the input.. which reduce it's purpose.

[a comment by an automated account is not shown]

**Comment by MysteryBlokHed** [comment]
What version of TypeScript and `@types/ramda` are you using? The example you linked seems to work for me

**Reply by assertnotnull** [reply]
@MysteryBlokHed Typescript 4.3.5 and @types/ramda 0.28.14
